posts/inside-rust/2022-07-29-compiler-team-2022-midyear-report.md

```diff
@@ -0,0 +1,800 @@
+
+For Jack Huey, switching to getting NLL stabilized was a nice change of pace. In a sense, it was “low-hanging fruit” and was a helpful mental break from pushing so hard on GATs for the past year.
+
+If Jack hadn’t been working on GATs for the past year or so, they would have instead been pushing harder on Chalk and librarifcation. In particular, there are fundamental questions, e.g. associated type normalization, to solve there. Recent work with a-mir-formality has started to help answer those. In the meantime, GATs were at a state that they were “unblocked”, had significant interest, and are a requirement for other language (async fns in traits) and lib (LendingIterator) features.
+
+### safe transmute
+
+<!-- https://hackmd.io/@rust-compiler-team/HJ7Y3s7uq -->
+
+**Progress:** The Rust project has made progress on this ambition since the start of the year
+
+**Goals:** We think this year’s planned goals for this ambition will be achieved in the next six months;
+we think the most important parts of the problem are now (or very soon to be) solved, and additional parts of the solution will be available in the next six months.
+
+**How it started:** At start of 2022, we had a plan, some of which had been completed, but more work remained to be done. Furthermore, that plan was not sufficient and was in need of revision before it could actually solve the problem.
+
+**How it's going:** we understand the problem better than we did at the start of the year,
+we have a high-level end-to-end plan to solve the problem,
+we have milestones that represent individual units of work for the near term that each make progress towards a solution
+we have Rust contributors who have agreed to help with the units of work that we have identified.
+Further more, we have completed some of our milestones; we think we have implemented a solution to the problem, but most of our work has not yet landed in hands of Rust users
+
+**Details:** At the start of the year, we opened [PR #92268](https://github.com/rust-lang/rust/pull/92268), *Initial Implementation of Transmutability Trait*, which aimed to provide the basic functionality of a trait implemented for any two types transmutable into each other (as defined by [MCP #411](https://github.com/rust-lang/compiler-team/issues/411)). This PR required additional testing and polish before it would be ready to merge, but progress unfortunately stalled in the spring.
+
+With the mentoring provided by [@oli-obk](https://github.com/oli-obk) and an influx of interest and help from [@m1el](https://github.com/m1el), progress resumed this summer; notably:
```

Comment:
Nit: I think we should either link everyone's usernames or no ones. Doing the easy thing for now but I can generate links if you think we should go that way. 

```suggestion
With the mentoring provided by @oli-obk and an influx of interest and help from @m1el, progress resumed this summer; notably:
```